src/main/java/org/bptlab/cepta/DataToDatabase.java

```diff
@@ -0,0 +1,159 @@
+  static final String USER = "tester";
+  static final String PASSWORD = "password";
+  static final Integer PORT = 5432;
+  private String table_name;
+  public DataToDatabase(String table){
+    this.table_name = table;
+  }
+  @Override
+  public Object map(Object dataSet) throws Exception {
+    insert(dataSet);
+    return dataSet;
+  }
+  public void insert(Object dataSet)
+      throws NoSuchFieldException, IllegalAccessException {
+
+    // Connection to PostgreSQL DB
+    ConnectionPool<PostgreSQLConnection> connection;
+
+    ConnectionPoolConfigurationBuilder config = new ConnectionPoolConfigurationBuilder();
+    config.setUsername(USER);
+    config.setPassword(PASSWORD);
+    config.setHost(HOST);
+    config.setPort(PORT);
+    config.setDatabase(DATABASE);
+    config.setMaxActiveConnections(100);
+    connection = PostgreSQLConnectionBuilder.createConnectionPool(config);
+
+    // stores strings of values and columns for sql query
+    String[] colsAndVals = columnsAndValuesToString(dataSet);
+
+    // Create query
+    String query = "INSERT INTO " + table_name + colsAndVals[0]
+        + " VALUES " + colsAndVals[1] + ";";
+    System.out.println(query);
+
+    // send query
+    CompletableFuture<QueryResult> future = connection.sendPreparedStatement(query);
+
+    // really execute query and get query result
+    try{
+      QueryResult result = future.get();
+    }catch (InterruptedException | ExecutionException e){
+      System.out.println("Could not get result");
+    }
+
+    // Close the connection pool
+    try{
+      connection.disconnect().get();
+    }catch (ExecutionException | InterruptedException e){
+      System.out.println("Could not disconnect");
+    }
+  }
+
+  private String[] columnsAndValuesToString(Object dataSet)
+      throws NoSuchFieldException, IllegalAccessException {
+    // takes the values matching to columns and converts them to
+    // a (val1, val2, ...) String
+    // necessary for usage in the sql statement
+
+    String valString = "(";
+    Class c = dataSet.getClass();
+
+    // get attribute fields of dataSet's class and store them in fields
+    Field[] fields = c.getDeclaredFields();
+
+    // stores names of attributes
+    String[] columns = new String[fields.length];
+/*
+    String[] columns = new String[fields.length];
+    for(int i = 0; i < fields.length; ++i){
+      columns[i] = fields[i].getName();
+    }
+*/
+
+    for(int i = 0; i < fields.length-1; i++){
+      // get attribute-field of class for column-name
+      Field f = fields[i];
+      f.setAccessible(true);
+
+      // add object's value of attribute/column to string
+      // with , in between
+      try{
+        if(f.getType().equals(" ".getClass())){
+          // add ' ' around value if it's a string
+          valString += "'" + f.get(dataSet).toString() + "'" + ", ";
+        }else{
+          valString += f.get(dataSet).toString() + ", ";
+        }
+        columns[i] = f.getName();
```

Comment:
Perhaps we could first read the data from the dataSet into its own variable and then use Java's instanceof operator to check whether or not its a String. 

A more elegant way of adding delimitors could be using [String's join method](https://www.javatpoint.com/java-string-join)

Reply:
Nice! That's a great way.